salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                              secret_engine.type, secret_engine.path)
+                else:
+                    log.debug(
+                        'Secret engine "%s" at path "%s" is not enabled. Enabling now...',
+                        secret_engine.type,
+                        secret_engine.path)
+
+                    client.sys.enable_secrets_engine(
+                        backend_type=secret_engine.type,
+                        path=secret_engine.path,
+                        description=secret_engine.description,
+                        config=secret_engine.config
+                    )
+
+                    new_secrets_engines.append("type: {} - path: {}".format(secret_engine.type, secret_engine.path))
+
+                    log.debug('Secret engine "%s" at path "%s" is enabled.',
+                              secret_engine.type, secret_engine.path)
+
+                if secret_engine.secret_config:
+                    log.info(
+                        'Provisioning specific configurations for "%s" secrets engine...', secret_engine.type)
+
+                    if secret_engine.type == 'ad':
+                        client.secrets.activedirectory.configure(
+                            **secret_engine.secret_config
+                        )
+                    if secret_engine.type == 'database':
+                        client.secrets.database.configure(
+                            **secret_engine.secret_config
+                        )
+
+                    log.info(
+                        'Finished provisioning specific configurations for "%s" secrets engine...', secret_engine.type)
+
+                if secret_engine.extra_config:
+                    log.info(
+                        'Provisioning extra conifgurations for for "%s" secrets engine...', secret_engine.type)
+
+                    if secret_engine.type == 'ad':
+                        # Get roles from vault
+                        existing_roles = None
+                        try:
+                            existing_roles = client.secrets.activedirectory.list_roles()
+                            log.debug(existing_roles)
+                        except Exception:
+                            raise
+
+                        # Add new roles
+                        local_roles = secret_engine.extra_config['roles']
+                        for key in local_roles:
+                            log.debug('AD Role ["%s"] -> Role %s',
+                                      str(key), local_roles[key])
+                            try:
+                                client.secrets.activedirectory.create_or_update_role(
+                                    name=key,
+                                    service_account_name=local_roles[key]['service_account_name'],
+                                    ttl=local_roles[key]['ttl']
+                                )
+                            except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.